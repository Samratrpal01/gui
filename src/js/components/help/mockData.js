export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
