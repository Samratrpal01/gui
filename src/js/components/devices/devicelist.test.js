import React from 'react';
import { prettyDOM } from '@testing-library/dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import DeviceList from './devicelist';

const mockStore = configureStore([thunk]);

describe('DeviceList Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });
  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <DeviceList
          devices={[]}
          deviceListState={defaultState.devices.deviceList}
          selectedRows={[]}
          columnHeaders={[{ name: 1 }, { name: 2 }, { name: 3 }, { name: 4 }]}
          pageTotal={50}
        />
      </Provider>
    );
    // special snapshot handling here to work around unstable ids in mui code...
    const view = prettyDOM(baseElement.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
