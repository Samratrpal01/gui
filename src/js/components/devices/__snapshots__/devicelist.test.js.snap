// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceList Component renders correctly 1`] = `
"<div>
  <div
    class=\\"deviceList \\"
    style=\\"grid-template-columns: 52px minmax(250px, 1fr) minmax(150px, 1.5fr) minmax(150px, 1.5fr) minmax(150px, max-content);\\"
  >
    <div
      class=\\"header\\"
    >
      <div
        class=\\"deviceListRow\\"
      >
        <div>
          <span
            class=\\"MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root\\"
          >
            <input
              checked=\\"\\"
              class=\\"PrivateSwitchBase-input css-1m9pwf3\\"
              data-indeterminate=\\"false\\"
              type=\\"checkbox\\"
            />
            <svg
              aria-hidden=\\"true\\"
              class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
              data-testid=\\"CheckBoxIcon\\"
              focusable=\\"false\\"
              viewBox=\\"0 0 24 24\\"
            >
              <path
                d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"
              />
            </svg>
            <span
              class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
            />
          </span>
        </div>
        <div
          class=\\"columnHeader flexbox space-between relative\\"
        >
          <div
            class=\\"flexbox center-aligned\\"
          />
          <div
            class=\\"flexbox center-aligned\\"
          />
        </div>
        <div
          class=\\"columnHeader flexbox space-between relative\\"
        >
          <div
            class=\\"flexbox center-aligned\\"
          />
          <div
            class=\\"flexbox center-aligned\\"
          >
            <span
              class=\\"resize-handle\\"
            />
          </div>
        </div>
        <div
          class=\\"columnHeader flexbox space-between relative\\"
        >
          <div
            class=\\"flexbox center-aligned\\"
          />
          <div
            class=\\"flexbox center-aligned\\"
          />
        </div>
        <div
          class=\\"columnHeader flexbox space-between relative\\"
        >
          <div
            class=\\"flexbox center-aligned\\"
          />
          <div
            class=\\"flexbox center-aligned\\"
          />
        </div>
      </div>
    </div>
    <div
      class=\\"body\\"
    />
    <div
      class=\\"footer flexbox margin-top\\"
    >
      <div
        class=\\"MuiTablePagination-root flexbox margin-top margin-top-none css-jtlhu6-MuiTablePagination-root\\"
      >
        <div
          class=\\"MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar\\"
        >
          <div
            class=\\"flexbox no-basis MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer\\"
          />
          <p
            class=\\"MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel\\"
            
          >
            Rows
          </p>
          <div
            class=\\"MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select\\"
            variant=\\"standard\\"
          >
            <div
              aria-expanded=\\"false\\"
              aria-haspopup=\\"listbox\\"
              
              class=\\"MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input\\"
              
              role=\\"button\\"
              tabindex=\\"0\\"
            >
              20
            </div>
            <input
              aria-hidden=\\"true\\"
              class=\\"MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput\\"
              tabindex=\\"-1\\"
              value=\\"20\\"
            />
            <svg
              aria-hidden=\\"true\\"
              class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon\\"
              data-testid=\\"ArrowDropDownIcon\\"
              focusable=\\"false\\"
              viewBox=\\"0 0 24 24\\"
            >
              <path
                d=\\"M7 10l5 5 5-5z\\"
              />
            </svg>
          </div>
          <p
            class=\\"MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows\\"
          />
          <div
            class=\\"flexbox center-aligned\\"
          >
            <div>
              1-20 of 50
            </div>
            <button
              class=\\"MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root\\"
              disabled=\\"\\"
              tabindex=\\"-1\\"
              type=\\"button\\"
            >
              <svg
                aria-hidden=\\"true\\"
                class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                data-testid=\\"KeyboardArrowLeftIcon\\"
                focusable=\\"false\\"
                viewBox=\\"0 0 24 24\\"
              >
                <path
                  d=\\"M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z\\"
                />
              </svg>
            </button>
            <div>
              <button
                class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root\\"
                tabindex=\\"0\\"
                type=\\"button\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                  data-testid=\\"KeyboardArrowRightIcon\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z\\"
                  />
                </svg>
                <span
                  class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
